app/controllers/api/plans_base_controller.rb

```diff
@@ -29,40 +33,38 @@ def authorize_action
   end
 
   def resource(id = params[:id])
-    return unless id.present?
+    return if id.blank?
+
     collection.readonly(false).find(id)
   end
 
   def collection
-    raise UndefinedCollectionMethod.new('You have to override collection method')
-  end
-
-  def find_plan
-    @plan = resource
+    raise UndefinedCollectionMethod, 'You have to override collection method'
   end
 
-  def find_plans
-    search = ThreeScale::Search.new(params[:search] || params)
-    @plans = collection.not_custom
-                       .order_by(params[:sort].presence || :name, params[:direction].presence || :asc)
-                       .scope_search(search)
-    @page_plans = @plans.paginate(pagination_params)
+  def plan_type
+    raise UndefinedCollectionMethod, 'You have to override plan_type method'
```

Comment:
The exception name is a bit confusing

Reply:
Yes, who knows who wrote that and when. Any naming suggestions? Would `UndefinedAbstractMethod` work?

Reply:
`UndefinedPlanTypeMethod` ? :upside_down_face:

Reply:
I would rather avoid defining 2 classes for 2 methods and do something more generic